src/wasm-type.h

```diff
@@ -24,7 +24,11 @@
 namespace wasm {
 
 class Type {
-  uint32_t id;
+  // enough for the limit of 1000 function arguments
+  static constexpr unsigned sizeBits = 10;
+  static constexpr unsigned unknownSize = (1 << sizeBits) - 1;
+  unsigned id : 32 - sizeBits;
```

Comment:
how about defining an `idBits = 32 - sizeBits` which could be used lower down too?

Reply:
Sounds good.